tests/test_utils.go

```diff
@@ -335,7 +336,31 @@ func removeURLSecrets(u *url.URL) string {
 
 // MatchInteraction checks if the request matches a store request in the given cassette.
 func MatchInteraction(r *http.Request, i cassette.Request) bool {
-	return r.Method == i.Method && removeURLSecrets(r.URL) == i.URL
+	// Default matching on method and URL without secrets
+	if !(r.Method == i.Method && removeURLSecrets(r.URL) == i.URL) {
+		return false
+	}
+
+	if r.Body == nil {
+		return true
+	}
+
+	var b bytes.Buffer
+	if _, err := b.ReadFrom(r.Body); err != nil {
+		return false
+	}
+	r.Body = ioutil.NopCloser(&b)
+
+	// Body must be empty or equal
+	if strings.HasPrefix(r.Header["Content-Type"][0], "multipart/form-data") {
+		fmt.Printf("skip %s ??? %s", b.String(), i.Body)
+		return true
+	}
+	matched := (b.String() == "" || b.String() == i.Body)
```

Comment:
why is it a match if the body is empty and different than the recorded body ?

Reply:
It's related to `GET` requests that do not contain body.

Reply:
But shouldn't it be equal to the recorded body `i.Body` anyway ?